packages/cli/src/lib/cmds/build.ts

```diff
@@ -30,137 +30,170 @@ interface SigningKeyPasswords {
+        signingKey.path,
+        passwords.keystorePassword, // keystore password
+        signingKey.alias, // alias
+        passwords.keyPassword, // key password
+        './app-release-unsigned-aligned.apk', // input file path
+        outputFile, // output file path
+    );
+    this.log.info(`Generated Android APK at "${outputFile}"`);
   }
 
-  const jdkHelper = new JdkHelper(process, config);
-  const androidSdkTools = new AndroidSdkTools(process, config, jdkHelper, log);
-  const keyTool = new KeyTool(jdkHelper, log);
-
-  if (!await androidSdkTools.checkBuildTools()) {
-    console.log('Installing Android Build Tools. Please, read and accept the license agreement');
-    await androidSdkTools.installBuildTools();
+  async buildAppBundle(signingKey: SigningKeyInfo, passwords: SigningKeyPasswords): Promise<void> {
+    await this.gradleWrapper.bundleRelease();
+    const inputFile = 'app/build/outputs/bundle/release/app-release.aab';
+    const outputFile = './app-release-bundle.aab';
+    await this.jarSigner.sign(
+        signingKey, passwords.keystorePassword, passwords.keyPassword, inputFile, outputFile);
+    this.log.info(`Generated Android App Bundle at "${outputFile}"`);
   }
 
-  const twaManifest = await TwaManifest.fromFile('./twa-manifest.json');
+  async build(): Promise<boolean> {
+    if (!await this.androidSdkTools.checkBuildTools()) {
+      console.log('Installing Android Build Tools. Please, read and accept the license agreement');
+      await this.androidSdkTools.installBuildTools();
+    }
 
-  const passwords = await getPasswords(log, prompt);
+    let validationPromise;
+    if (!this.args.skipPwaValidation) {
+      validationPromise = this.runValidation();
+    }
 
-  // Builds the Android Studio Project
-  log.info('Building the Android App...');
-  const gradleWraper = new GradleWrapper(process, androidSdkTools);
-  await gradleWraper.assembleRelease();
+    const twaManifest = await TwaManifest.fromFile('./twa-manifest.json');
+    const passwords = await this.getPasswords();
 
-  // Zip Align
-  log.info('Zip Aligning...');
-  await androidSdkTools.zipalign(
-      './app/build/outputs/apk/release/app-release-unsigned.apk', // input file
-      './app-release-unsigned-aligned.apk', // output file
-  );
+    // Builds the Android Studio Project
+    this.log.info('Building the Android App...');
+    await this.buildApk(twaManifest.signingKey, passwords);
 
-  if (!args.skipPwaValidation) {
-    log.info('Checking PWA Quality Criteria...');
-    try {
-      const pwaValidationResult = (await pwaValidationPromise)!;
-      printValidationResult(pwaValidationResult, log);
-      if (pwaValidationResult.status === 'FAIL') {
-        log.warn('PWA Quality Criteria check failed.');
+    if (this.args.generateAppBundle) {
+      await this.buildAppBundle(twaManifest.signingKey, passwords);
+    }
+
+    await this.generateAssetLinks(twaManifest, passwords);
+
+    if (validationPromise) {
```

Comment:
Could you make this line a bit clearer - are we checking that the validationPromise is not null, or are we checking that it is not false?

Reply:
done - Changed to declare `let validationPromise = null` earlier so we can check `validationPromise !== null` here.